### PR TITLE
Add check for IPA KRA Agent

### PR DESCRIPTION
**Pull request by antoniotorresm**
Add check to validate KRA Agent in case KRA is installed, including
checking for the KRA Agent LDAP entry.

Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1894781
Signed-off-by: Antonio Torres <antorres@redhat.com>

### CODE REVIEW

**Comment by rcritten** [comment]
I think a better approach would to pull much of this out into a function that takes the DN to tell which entry to compare and return so a Result can be constructed. The is_installed check for the CA and RA/KRA will probably have to remain in the check itself for simplicity.

This will reduce a lot of code duplication.

A test would be nice too.

File: src/ipahealthcheck/ipa/certs.py [hunk 1]
```diff
@@ -724,6 +724,81 @@ def check(self):
                         self, constants.SUCCESS, key=cert)
 
 
+def check_agent(plugin, base_dn):
```
Comment (rcritten) [inline src/ipahealthcheck/ipa/certs.py:727]: If it took a string you could differentiate agent type:

e.g check_agent(plugin, base_dn, type):
...
logger.debug('%s agent descripiton should be %s', type, description)
...
        yield Result(plugin, constants.ERROR,
                     description=description,
                     msg='%s agent not found in LDAP' % type)

I guess it's implicit so this isn't a hard ask but it does read better IMHO, particularly in the human output.

**Comment by rcritten** [comment]
Thinking about this further, I really think that the message should include KRA or RA for readability purposes.

**Comment by antoniotorresm** [comment]
Yes, I added it as you suggested :)

**Comment by rcritten** [comment]
Huh, I must have not reloaded the patch or something. In that case ack.

**Pull request merged**